src/lib/logical_query_plan/static_table_node.hpp

```diff
@@ -24,6 +24,8 @@ class StaticTableNode : public EnableMakeForLQPNode<StaticTableNode>, public Bas
 
  protected:
   mutable std::optional<std::vector<std::shared_ptr<AbstractExpression>>> _column_expressions;
+  // static table nodes that reference meta tables should not be copied
+  const bool _copyable;
```

Comment:
Suggestion: "Some tables should not be copied but recreated. Currently, this applies to meta tables."

Also, I'd move that to the constructor (which has no documentation on that parameter right now)